File: room-controller/room-controller/src/Sensors.cpp
#include "Sensors.h"

Sensors::Sensors()
{
	this->pir = false;
	this->photoresistor = 0;
}

/**
 * @return true if the pir is on, false otherwise
 */
bool Sensors::isPirOn()
{
	return pir;
}

/**
 * @return true if the photoresistor value is greater than lightLimit. This parameter can be
 * specified in the constructor
 */
bool Sensors::isLightLow()
{
	return this->photoresistor <= THL;
}

/**
 * Update the status of the pir and the value of the photoresistor
 */
void Sensors::update(bool pirStatus, int photoresistorValue)
{
	setPirStatus(pirStatus);
	setPhotoresistorValue(photoresistorValue);
}

/**
 * @param pirStatus the new status of the pir. True if someone i detected in the room,
 *  false otherwise
 */
void Sensors::setPirStatus(bool pirStatus)
{
	this->pir = pirStatus;
}

/**
 * @param photoresistorValue the new value detected by the photoresistor
 */
void Sensors::setPhotoresistorValue(int photoresistorValue)
{
	this->photoresistor = photoresistorValue;
}

File: room-service/src/sensors/Sensors.java
package sensors;

public class Sensors {

	private boolean pir; 
	private int photoresistor; 

	
	public Sensors() {
		this.pir = false; 
		this.photoresistor = 0;  
	}

	/**
	 * 
	 * @return true if the pir is on, false otherwise
	 */
	public boolean isPirOn() {
		return pir; 
	}
	
	public int getPhotoresistorValue() {
		return this.photoresistor; 
	}
	

	/**
	 * 
	 * @param pirStatus the new status of the pir. True if someone i detected in the room,
	 *  false otherwise
	 */
	public void setPirStatus(boolean pirStatus) { 
		this.pir = pirStatus; 
	}
	
	/**
	 * 
	 * @param photoresistorValue the new value detected by the photoresistor
	 */
	public void setPhotoresistorValue(int photoresistorValue) {
		this.photoresistor = photoresistorValue; 
	}

	@Override
	public String toString() {
		return "Sensors [pir=" + pir + ", photoresistor=" + photoresistor + "]";
	}
		
}
